File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        Map<String, Integer> participantMap = new HashMap<>();
		
		for(String part : participant) {
			participantMap.put(part, participantMap.getOrDefault(part, 0) + 1);
		}
		
		for(String comp : completion) {
			participantMap.put(comp, participantMap.get(comp) - 1);
		}
		
		for(String key : participantMap.keySet()) {
			if(participantMap.get(key) == 1) {
				answer = key;
			} else {
				continue;
			}
		}
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    let map = new Map();
    for (let part of participant){
        if(map.has(part)) map.set(part, map.get(part) + 1);
        else map.set(part, 1);
    }
    for (let comp of completion){
        if (map.get(comp) === 1) map.delete(comp);
        else map.set(comp, map.get(comp) - 1);
    }
    for (m of map.keys()) return m;
}